### ISSUE TITLE
training Inceptionv2+SSD

### ISSUE THREAD

**Issue by jillelajitta** [algorithmiaio/sample-apps#15]
Hi @zeryx and @hmahyar 

I have read your post https://blog.algorithmia.com/deep-dive-into-object-detection-with-open-images-using-tensorflow/. I have implemented  faster_rcnn_resnet101 using your instructions. It is working super cool and your post is amazing, very clear.

I tried implementing  SSD+inceptionV2 using your config file on OPEN IMAGES DATASET. But I'm not sure for how many steps to train the model and can I use COCO Inception v2 pretrained weights?

**Comment by zeryx** [comment]
Thanks so much for the kind words @jillelajitta :smile: 
Regarding SSD + inceptionV2, you can use the inceptionv2 weights with minimal changes!
In regards to training steps (I'm assuming you're talking epochs?), the real value doesn't matter here as much - due to how the object detection pipeline's training data synthesis system it's very difficult to overtrain, which means it doesn't really matter how many steps are "too many" during training.
The big thing here is to ensure that your `eval.py` script is running, and that you are measuring a real error reduction between epochs on the test set. We'd recommend holding out additional data for `validation`, so you can make sure that you're not overfitting.

Let me know if that helps! Happy you found our tutorial useful :grin: 
James

**Comment by zeryx** [comment]
1) No that shouldn't cause any problems at all, actually it should make it easier to fine tune.
2) I think the issue here is the design of the Single Shot Detector architecture itself, in how it draws boxes. The output layer for such a network is quite wide and sparse, which may lead to training difficulties as they increase loss surface complexity. In a world where there were no class imbalances at all, this would probably be pretty easy to deal with - but as you know the OID has some significant class imbalances.
3) If you want to isolate the problem I'd recommend focusing on a very small but common subset, maybe just focus on `Person` and see what the results are for that, and then move on from there?

**Comment by jillelajitta** [comment]
That's amazing reply, very clear. What do you mean by class imbalances? No.of.images for each class is not same? Also I wonder, is my scheduled learning rate values are correct?

**Comment by zeryx** [comment]
1,2) Yea exactly, in the OID there are some classes that are heavily represented aka more images, like people and cars, and others that are very underrepresented - much less images. The underrepresented classes can affect your overall mAP and bring down the accuracy of your model overall.
3) I haven't adjusted the learning rate parameters from the defaults as they didn't seem to have any direct effect (a 0.5% mAP difference is not enough to justify anything for me), however if you find a combo that works better I'd love to know about it!

**Comment by jillelajitta** [comment]
Sure I will let you know if I find a good combo.

Now I will work on dataset to make all images into equal number for all classes. Thank you so much.

**Comment by zeryx** [comment]
Closing for now, feel free to reopen if you have further questions :grin:
